pkg/virt-handler/vm.go

```diff
@@ -329,35 +331,35 @@ func (c *VirtualMachineController) execute(key string) error {
 	if err != nil {
 		return err
 	}
-	c.logger.Object(vmi).V(4).Infof("domain exists %v", domainExists)
+	c.logger.Object(vmiCopy).V(4).Infof("domain exists %v", domainExists)
 
 	if !vmiExists && string(domainCachedUID) != "" {
 		// it's possible to discover the UID from cache even if the domain
 		// doesn't technically exist anymore
-		vmi.UID = domainCachedUID
-		c.logger.Object(vmi).Infof("Using cached UID for vmi found in domain cache")
```

Comment:
Actually this is guarded by `vmiExists` so it cannot be from cache. We should be able to unit test this.